### ISSUE TITLE
How to enable HTTPS ?

### ISSUE THREAD

**Issue by joce007** [BALKANGraph/export-service#3]
Hi,

I am currently evaluating the OrgChartJS library including the export-service. It is a really great tool including the export-service and we are looking for buying licenses. Therefore, I have a hard time setting the export-service to support HTTPS requests.  I am not sure if this is the same issue as describe in this post: https://github.com/balkangraph/orgchartjs/issues/289 but I don't see how the fix could solve my issue. 

I've tried to use the lastest source code and I am not able to make it works. Not sure if I'm doing this right or misunderstands how it should works.  Based on the source code of https://github.com/BALKANGraph/export-service/blob/master/app.js, I don't see where we enable HTTPS on Express to support it.  And also where the certificates and keys are extracted ?  

I would have assume seeing something like this:

```
const express = require('express');
const app = express();
var http = require('http');
var https = require('https');
var privateKey  = fs.readFileSync('server.key', 'utf8');
var certificate = fs.readFileSync('server.crt', 'utf8');
var credentials = {key: privateKey, cert: certificate};

var httpServer = http.createServer(app);
var httpsServer = https.createServer(credentials, app);
httpServer.listen(port);
httpsServer.listen(sslport);
```
I would need advice on how to set this up!  Without this capability, it would be hard for us to recommend this tool.  

Thanks!

**Comment by plamen-peshev** [comment]
Do you have certificate files server.key and  server.crt?

**Comment by joce007** [comment]
@plamen-peshev Yes I have them, They could be generated easily using openssl as follow:

`openssl req -nodes -new -x509 -keyout server.key -out server.cert`

**Comment by zabuzapr** [comment]
> Hi,
> 
> I am currently evaluating the OrgChartJS library including the export-service. It is a really great tool including the export-service and we are looking for buying licenses. Therefore, I have a hard time setting the export-service to support HTTPS requests. I am not sure if this is the same issue as describe in this post: [BALKANGraph/OrgChartJS#289](https://github.com/BALKANGraph/OrgChartJS/issues/289) but I don't see how the fix could solve my issue.
> 
> I've tried to use the lastest source code and I am not able to make it works. Not sure if I'm doing this right or misunderstands how it should works. Based on the source code of https://github.com/BALKANGraph/export-service/blob/master/app.js, I don't see where we enable HTTPS on Express to support it. And also where the certificates and keys are extracted ?
> 
> I would have assume seeing something like this:
> 
> ```
> const express = require('express');
> const app = express();
> var http = require('http');
> var https = require('https');
> var privateKey  = fs.readFileSync('server.key', 'utf8');
> var certificate = fs.readFileSync('server.crt', 'utf8');
> var credentials = {key: privateKey, cert: certificate};
> 
> var httpServer = http.createServer(app);
> var httpsServer = https.createServer(credentials, app);
> httpServer.listen(port);
> httpsServer.listen(sslport);
> ```
> 
> I would need advice on how to set this up! Without this capability, it would be hard for us to recommend this tool.
> 
> Thanks!

**Comment by zabuzapr** [comment]
> Hi,
> 
> I am currently evaluating the OrgChartJS library including the export-service. It is a really great tool including the export-service and we are looking for buying licenses. Therefore, I have a hard time setting the export-service to support HTTPS requests. I am not sure if this is the same issue as describe in this post: [BALKANGraph/OrgChartJS#289](https://github.com/BALKANGraph/OrgChartJS/issues/289) but I don't see how the fix could solve my issue.
> 
> I've tried to use the lastest source code and I am not able to make it works. Not sure if I'm doing this right or misunderstands how it should works. Based on the source code of https://github.com/BALKANGraph/export-service/blob/master/app.js, I don't see where we enable HTTPS on Express to support it. And also where the certificates and keys are extracted ?
> 
> I would have assume seeing something like this:
> 
> ```
> const express = require('express');
> const app = express();
> var http = require('http');
> var https = require('https');
> var privateKey  = fs.readFileSync('server.key', 'utf8');
> var certificate = fs.readFileSync('server.crt', 'utf8');
> var credentials = {key: privateKey, cert: certificate};
> 
> var httpServer = http.createServer(app);
> var httpsServer = https.createServer(credentials, app);
> httpServer.listen(port);
> httpsServer.listen(sslport);
> ```
> 
> I would need advice on how to set this up! Without this capability, it would be hard for us to recommend this tool.
> 
> Thanks!

Hi joyce, 

it seems i'm facing the same problem, how do you solve this problem? since it's not closed
